tests/Unit/Evolution/Systems/GeneralizedHarmonic/TestFunctions.py

```diff
@@ -184,3 +183,60 @@ def two_index_constraint(d_gauge_function, spacetime_normal_one_form,
     return constraint
 
 # End test functions for two-index constraint
+
+
+# Test functions for characteristic speeds
+def char_speed_upsi(gamma1, lapse, shift, unit_normal):
+    return - (1. + gamma1) * np.dot(shift, unit_normal)
+
+
+def char_speed_uzero(gamma1, lapse, shift, unit_normal):
+    return - np.dot(shift, unit_normal)
+
+
+def char_speed_uplus(gamma1, lapse, shift, unit_normal):
+    return - np.dot(shift, unit_normal) + lapse
+
+
+def char_speed_uminus(gamma1, lapse, shift, unit_normal):
+    return - np.dot(shift, unit_normal) - lapse
+
+# End test functions for characteristic speeds
+
+
+# Test functions for characteristic fields
```

Comment:
autopep8 (python's analog of clang-format)

Reply:
changed